ios/browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.mm

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.h"
+
+#include <string>
+#include <utility>
+
+#include "base/notimplemented.h"
+#include "base/strings/escape.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_service.h"
+#include "brave/components/ai_chat/core/browser/constants.h"
+#include "brave/components/ai_chat/core/browser/conversation_handler.h"
+#include "brave/components/ai_chat/core/common/ai_chat_urls.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/common/mojom/untrusted_frame.mojom.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "brave/components/ai_chat/core/common/prefs.h"
+#include "brave/components/ai_chat/resources/grit/ai_chat_ui_generated_map.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/ios/browser/ai_chat/ai_chat_service_factory.h"
+#include "brave/ios/browser/ui/webui/ai_chat/ai_chat_ui.h"
+#include "brave/ios/browser/ui/webui/favicon_source.h"
+#include "brave/ios/browser/ui/webui/untrusted_sanitized_image_source.h"
+#include "brave/ios/web/webui/brave_url_data_source_ios.h"
+#include "brave/ios/web/webui/brave_web_ui_ios_data_source.h"
+#include "brave/ios/web/webui/brave_webui_utils.h"
+#include "components/favicon_base/favicon_url_parser.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/grit/brave_components_webui_strings.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "ios/chrome/browser/shared/model/profile/profile_ios.h"
+#include "ios/components/webui/web_ui_url_constants.h"
+#include "ios/web/public/web_state.h"
+#include "ios/web/public/webui/url_data_source_ios.h"
+#include "ios/web/public/webui/web_ui_ios.h"
+#include "ios/web/public/webui/web_ui_ios_data_source.h"
+#include "ios/web/web_state/web_state_impl.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote.h"
+#include "third_party/abseil-cpp/absl/strings/str_format.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/webui/webui_util.h"
+#include "url/url_constants.h"
+
+namespace {
+constexpr char kURLLearnMoreBraveSearchLeo[] =
```

Comment:
Can probably share this with desktop/android

Reply:
cc @fallaciousreasoning @petemill

Reply:
Yeah, should probably live in components/ai_chat/core/common/ai_chat_urls.h now you've made it available on ios :smile:

Reply:
opened PR to move them here: https://github.com/brave/brave-core/pull/31455